src/utils/mod.rs

```diff
@@ -63,6 +63,28 @@ pub(crate) fn expand_bits(bit_depth: u8, row_size: u32, buf: &[u8]) -> Vec<u8> {
     p
 }
 
+/// Checks if the provided dimensions would cause an overflow.
+/// 
+/// Returns the optionally provided `ImageError` on overflow or defaults to `LimitErrorKind::DimensionError`.
+#[allow(dead_code)]
+// When no image formats that use it are enabled
+pub(crate) fn check_dimension_overflow(
+    width: u32,
+    height: u32,
+    bytes_per_pixel: u8,
+    error: Option<crate::ImageError>,
+) -> crate::ImageResult<()> {
```

Comment:
It might be better to have this function just return a bool, and make the caller deal with throwing the error.

Reply:
Hm, also a good idea to avoid the issue [mentioned here](https://github.com/image-rs/image/pull/1407#discussion_r563164600). Either way is fine for me.